Sources/NIOCore/EventLoop.swift

```diff
@@ -317,6 +317,10 @@ public protocol EventLoop: EventLoopGroup {
     /// implementation returns a ``NIODefaultSerialEventLoopExecutor`` instead, which provides suboptimal performance.
     @available(macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0, *)
     var executor: any SerialExecutor { get }
+
+    /// Submit a job to be executed by the `EventLoop`
+    @available(macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0, *)
+    func enqueue(_ job: consuming ExecutorJob)
```

Comment:
I don't think we can add the consuming keyword here without compiler guards.

Reply:
This is compiler guarded